src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -352,67 +352,60 @@ public BigInteger(ReadOnlySpan<byte> value, bool isUnsigned = false, bool isBigE
             }
             else
             {
-                int unalignedBytes = byteCount % 4;
-                int dwordCount = byteCount / 4 + (unalignedBytes == 0 ? 0 : 1);
-                uint[] val = new uint[dwordCount];
-                int byteCountMinus1 = byteCount - 1;
-
-                // Copy all dwords, except don't do the last one if it's not a full four bytes
-                int curDword, curByte;
+                int wholeDwordCount = Math.DivRem(byteCount, 4, out int unalignedBytes);
```

Comment:
Should we use:
```C#
(uint wholeUInt32Count, uint unalignedBytes) = Math.DivRem((uint)byteCount, 4);
```
instead, since we know here byteCount isn't negative?

Reply:
If we did, then we would have to cast the values back to int in each of the `AsSpan` and `Slice` calls. What do you suggest?